cli/util/api.go

```diff
@@ -231,7 +232,7 @@ type contextKey string
 
 // OnSingleNode is not thread safe
```

Comment:
```suggestion
// OnSingleNode is thread safe
```

Reply:
Ok, honestly, that makes me kind of angry.

1. `OnSingleNode` _was_ threadsafe.
2. Using a context derived from it was not.
3. This was horribly dangerous. And by design? REALLY?!?